test/watch-workspace-item.test.js

```diff
@@ -0,0 +1,203 @@
+        assert.isTrue(component.state.someKey);
+      });
+    });
+  });
+
+  describe('workspace events', function() {
+    it('becomes true when the pane is opened', async function() {
+      sub = watchWorkspaceItem(workspace, 'atom-github://item/{pattern}', component, 'theKey');
+
+      assert.isFalse(component.state.theKey);
+
+      await workspace.open('atom-github://item/match');
+
+      assert.isTrue(component.setState.calledWith({theKey: true}));
+    });
+
+    it('remains true if another matching pane is opened', async function() {
+      await workspace.open('atom-github://item/match0');
+      sub = watchWorkspaceItem(workspace, 'atom-github://item/{pattern}', component, 'theKey');
+
+      assert.isTrue(component.state.theKey);
+
+      await workspace.open('atom-github://item/match1');
+
+      assert.isFalse(component.setState.called);
+    });
+
+    it('remains true if a matching pane is closed but another remains open', async function() {
+      await workspace.open('atom-github://item/match0');
+      await workspace.open('atom-github://item/match1');
+
+      sub = watchWorkspaceItem(workspace, 'atom-github://item/{pattern}', component, 'theKey');
+      assert.isTrue(component.state.theKey);
+
+      assert.isTrue(workspace.hide('atom-github://item/match1'));
+
+      assert.isFalse(component.setState.called);
+    });
+
+    it('becomes false if the last matching pane is closed', async function() {
+      await workspace.open('atom-github://item/match0');
+      await workspace.open('atom-github://item/match1');
+
+      sub = watchWorkspaceItem(workspace, 'atom-github://item/{pattern}', component, 'theKey');
+      assert.isTrue(component.state.theKey);
+
+      assert.isTrue(workspace.hide('atom-github://item/match1'));
+      assert.isTrue(workspace.hide('atom-github://item/match0'));
+
+      assert.isTrue(component.setState.calledWith({theKey: false}));
+    });
+
+    describe('{active: true}', function() {
+      //
+    });
+  });
+
+  it('stops updating when disposed', async function() {
+    sub = watchWorkspaceItem(workspace, 'atom-github://item', component, 'theKey');
+    assert.isFalse(component.state.theKey);
+
+    sub.dispose();
+    await workspace.open('atom-github://item');
+    assert.isFalse(component.setState.called);
+
+    await workspace.hide('atom-github://item');
+    assert.isFalse(component.setState.called);
+  });
+
+  describe('setPattern', function() {
+    it('immediately updates the state based on the new pattern', async function() {
+      sub = watchWorkspaceItem(workspace, 'atom-github://item0/{pattern}', component, 'theKey');
+      assert.isFalse(component.state.theKey);
+
+      await workspace.open('atom-github://item1/match');
+      assert.isFalse(component.setState.called);
+
+      await sub.setPattern('atom-github://item1/{pattern}');
+      assert.isFalse(component.state.theKey);
```

Comment:
Nope! `component` isn't a real React component, just a mock one, so the `.state` property and `.setState()` method are (intentionally) unrelated. What I'm testing here is that `setPattern()` is using `component.setState()` to modify the "component's" state with the new pattern instead of directly setting it on `component.state`.